lib/Models/Segment.php

```diff
@@ -155,6 +155,24 @@ static function getWooCommerceSegment() {
     return $wc_segment;
   }
 
+  static function shouldShowWooCommerceSegment() {
+    $is_woocommerce_active = class_exists('WooCommerce');
+    $woocommerce_users_count = Subscriber::where('is_woocommerce_user', 1)->count();
```

Comment:
For big segments it would be more performant to use sth like "WHERE is_woocommerce_user = 1 LIMIT 1", it also seems `is_woocommerce_user` doesn't have index but this should help.

Btw. I also checked the option of looking to the segment directly and sth like `SubscriberSegment::where('segment_id', $wc_segment_id)->limit(1)` should work if you know the segment ID.